homeassistant/components/notify/html5.py

```diff
@@ -129,7 +180,108 @@ def post(self, request):
+        # 3.  If no target, loop through all registrations & attempt decode.
+        # 3a. If a registration successfully decodes, return the payload.
+        # 4.  Unable to decode the JWT, return False.
+
+        target_check = jwt.decode(token, verify=False)
+        if target_check[ATTR_TARGET] in self.registrations.keys():
+            possible_target = self.registrations[target_check[ATTR_TARGET]]
+            key = possible_target[ATTR_SUBSCRIPTION][ATTR_KEYS][ATTR_AUTH]
+            try:
+                return jwt.decode(token, key)
+            except jwt.exceptions.DecodeError:
+                pass
+
+        for reg in self.registrations.values():
+            try:
+                key = reg[ATTR_SUBSCRIPTION][ATTR_KEYS][ATTR_AUTH]
+                return jwt.decode(token, key)
+            except jwt.exceptions.DecodeError:
+                continue
+        return False
+
+    # The following is based on code from Auth0
+    # https://auth0.com/docs/quickstart/backend/python
+    # pylint: disable=too-many-return-statements
+    def check_authorization_header(self, request):
+        """Check the authorization header."""
+        import jwt
+        auth = request.headers.get('Authorization', None)
+        if not auth:
+            return self.json_message('Authorization header is expected',
+                                     status_code=HTTP_UNAUTHORIZED)
+
+        parts = auth.split()
+
+        if parts[0].lower() != 'bearer':
+            return self.json_message('Authorization header must '
+                                     'start with Bearer',
+                                     status_code=HTTP_UNAUTHORIZED)
+        elif len(parts) != 2:
+            return self.json_message('Authorization header must '
+                                     'be Bearer token',
+                                     status_code=HTTP_UNAUTHORIZED)
+
+        token = parts[1]
+        try:
+            payload = self.decode_jwt(token)
+        except jwt.exceptions.InvalidTokenError:
+            return self.json_message('token is invalid',
+                                     status_code=HTTP_UNAUTHORIZED)
+        return payload
+
+    def post(self, request):
+        """Accept the POST request for push registrations event callback."""
+        auth_check = self.check_authorization_header(request)
+        if not isinstance(auth_check, dict):
+            return auth_check
+
+        event_payload = {
+            ATTR_TAG: request.json.get(ATTR_TAG),
+            ATTR_TYPE: request.json[ATTR_TYPE],
+            ATTR_TARGET: auth_check[ATTR_TARGET],
+        }
+
+        if request.json.get(ATTR_ACTION) is not None:
+            event_payload[ATTR_ACTION] = request.json.get(ATTR_ACTION)
+
+        if request.json.get(ATTR_DATA) is not None:
+            event_payload[ATTR_DATA] = request.json.get(ATTR_DATA)
+
+        try:
+            CALLBACK_EVENT_PAYLOAD_SCHEMA(event_payload)
```

Comment:
As sometimes schemas can change data, it's good practice to do `payload = CALLBACK_…(event_payload)`